drivers/ws281x/ws281x_saul.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ws281x
+ * @{
+ *
+ * @file
+ * @brief       NEOPixel/ws281x adaption to SAUL
+ *
+ * @author      Kevin Weiss <kevin.weiss@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include "saul.h"
+#include "ws281x.h"
+
+static int set_rgb_led(const void *dev, const phydat_t *res)
+{
+    ws281x_t *ws281x = (ws281x_t *)dev;
+    color_rgb_t color = {
+        .r = res->val[0],
+        .g = res->val[1],
+        .b = res->val[2]
+    };
+    for (int i = 0; i < ws281x->params.numof; i++) {
```

Comment:
Here you set all LEDs, but in auto-init you create a SAUL entry for every individual LED